squad/api/rest.py

```diff
@@ -784,31 +785,60 @@ class EnvironmentViewSet(ModelViewSet):
 
 
 class HyperlinkedMetricsIdentityField(serializers.HyperlinkedIdentityField):
+
     def get_url(self, *args):
         testrun = args[0]
         statuses = testrun.status.all()
         if len(statuses) > 0:
-            if testrun.status.all()[0].has_metrics:
-                return super().get_url(*args)
+            for s in statuses:
+                if not s.suite_id and s.has_metrics:
+                    return super().get_url(*args)
+                else:
+                    return None
         else:
             return None
 
 
 class HyperlinkedTestsIdentityField(serializers.HyperlinkedIdentityField):
+
     def get_url(self, *args):
         testrun = args[0]
         statuses = testrun.status.all()
         if len(statuses) > 0:
-            tr_status = testrun.status.all()[0]
-            num_tests = (
-                tr_status.tests_pass + tr_status.tests_fail + tr_status.tests_skip + tr_status.tests_xfail
-            )
-            if num_tests > 0:
-                return super().get_url(*args)
+            for s in statuses:
+                if not s.suite_id:
+                    num_tests = (s.tests_pass + s.tests_fail + s.tests_skip + s.tests_xfail)
+                    if num_tests > 0:
+                        return super().get_url(*args)
+                    else:
+                        return None
         else:
             return None
 
 
+class StatusFilteredListSerializer(serializers.ListSerializer):
```

Comment:
maybe we should use drf-extensions instead? This implementation looks a bit overcomplicated to me. drf-extensions: https://github.com/chibisov/drf-extensions